src/main/java/fi/aalto/cs/apluscourses/presentation/filter/Options.java

```diff
@@ -31,4 +33,29 @@ protected void selectionChanged() {
   public Iterator<Option> iterator() {
     return optionList.iterator();
   }
+
+  public void toggleAll() {
+    boolean isAnyActive = isAnyActive();
+    optionList.forEach(option -> option.isSelected.set(isAnyActive));
+  }
+
+  /**
+   * If any filter is active, returns true.
+   */
+  public boolean isAnyActive() {
+    return optionList.stream()
+            .anyMatch(option -> {
+              var selected = option.isSelected.get();
+              return selected != null && !selected;
```

Comment:
For future reference: `Boolean.FALSE.equals(option.isSelected.get())` does the null-check for you :wink: